mariadb/kontena.yml

```diff
@@ -0,0 +1,83 @@
+stack: kontena/mariadb
+version: 0.2.0
+expose: server
+variables:
+  root_password:
+    type: string
+    from:
+      vault: ${STACK}-root-password
+      random_string: 16
+    to:
+      vault: ${STACK}-root-password
+  user:
+    type: string
+    default: mariadb
+    from:
+      vault: ${STACK}-user
+      prompt: Username
+    to:
+      vault: ${STACK}-user
+  password:
+    type: string
+    from:
+      vault: ${STACK}-password
+      random_string: 16
+    to:
+      vault: ${STACK}-password
+  db:
+    type: string
+    default: mariadb
+    from:
+      vault: ${STACK}-database
+      prompt: Database name
+    to:
+      vault: ${STACK}-database
+  affinity:
+    type: string
+    default: label==~mariadb
+    from:
+      prompt: Affinity
+  mem_limit:
+    type: integer
+    default: 512
+    from:
+      prompt: Memory limit (MB)
+  innodb_buffer_pool_size:
+    type: string
+    from:
+      evaluate: ((${mem_limit} * 10) * 7) / 100
+services:
+  server:
+    image: mariadb:10.2
+    instances: 1
+    command: "mysqld --innodb-buffer-pool-size={{ innodb_buffer_pool_size }}M"
+    mem_limit: "{{ mem_limit }}m"
+    stateful: true
```

Comment:
Why both `stateful: true` and volumes for the data?

Reply:
Stateful affects also how scheduler behaves. Of course it can be removed but it's not just for volumes here.